### PR TITLE
[WNMGDS-1273] Convert Healthcare Footer to TypeScript

### PR DESCRIPTION
**Pull request by zarahzachz**
## Summary
https://jira.cms.gov/browse/WNMGDS-1273

### Changed
- Convert HC.gov Footer and all subcomponents to TypeScript

### Fixed
- Update the i18next-related code to be compatible with its new API
  - Unfortunately this means perhaps misusing its namespaces feature in order for us to be able to keep our own API unchanged. I have made a note about it in the code.

## How to test
View the storybook story for the footer to test it. Make sure switching between English and Spanish works.

### CODE REVIEW

File: packages/ds-healthcare-gov/src/components/i18n.js [hunk 1]
```diff
@@ -7,6 +7,18 @@ import en from './locale/en.json';
...
+  // The keys of this resources object are supposed to be the languages, but we're using
+  // the namespaces feature as our language switcher because that will allow us to continue
+  // to set the language on a per-component basis. Otherwise, after upgrading to the latest
+  // version of i18next we'd have to start setting the language at the i18nInstance level.
+  common: {
+    en,
+    es,
+  },
+};
+
```
Comment (pwolfert) [inline packages/ds-healthcare-gov/src/components/i18n.js:21]: Here's the funny business

**Comment by zarahzachz** [comment]
Running tests locally produces some errors for me:
```
console.error
      Error: Not implemented: window.scroll
```

**Comment by pwolfert** [comment]
@zarahzachz, I saw that but wasn't sure how to resolve it. I figured after some googling that it was an outdated version of jsdom, but I think it just might not have been implemented. I can try mocking it in the tests setup file

**Comment by pwolfert** [comment]
@zarahzachz, I added a mock function

**Review by forrestbaer — APPROVED** [review]
tests passed, types passed, looks good all around

**Pull request merged**